src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -740,6 +750,7 @@ internal class LoggerParameter
             public string Name = string.Empty;
             public string Type = string.Empty;
             public string? Qualifier;
+            public bool NeedsAtSign;
```

Comment:
I wonder if it would be better if we created a new field/property that is the "CodeIdentifier" or "CodeName" or similar that all the places in the Emitter could use. This field would have `@` in it if needed. Then all the places in the Emitter wouldn't need to do the checking - they would just call this new field/property.